### ISSUE TITLE
ClickHouse dialect does not support table creation syntax

### ISSUE THREAD

**Issue by jbradt** [sqlfluff/sqlfluff#3505]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff fails to parse ClickHouse `create table` statements that use non-standard syntax for providing engine options.

### Expected Behaviour

The tool should be able to parse these statements.

### Observed Behaviour

A creation query like this one:

```sql
create table example1
(
    a String,
    b String
)
engine = MergeTree()
order by (a, b);
```

Gives errors
```
❯ sqlfluff lint example.sql --dialect clickhouse
== [example.sql] FAIL
L:   5 | P:   2 |  PRS | Line 5, Position 2: Found unparsable section: '\nengine
                       | = MergeTree()\norder by (a, b)'
L:  21 | P:   1 | L009 | Files must end with a single trailing newline.
WARNING: Parsing errors found and dialect is set to 'clickhouse'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

Please note that the query above doesn't include all possible options for the table creation statement. [This page](https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree/#table_engine-mergetree-creating-a-table) in the ClickHouse docs shows a more complete description of the syntax. 

### How to reproduce

```sql
-- example.sql
create table example1
(
    a String,
    b String
)
engine = MergeTree()
order by (a, b);
```

Command:

```
sqlfluff lint example.sql --dialect clickhouse
```

### Dialect

ClickHouse

### Version

```
❯ sqlfluff --version
sqlfluff, version 1.0.0

❯ python --version
Python 3.7.13
```

### Configuration

This was just with the stock configuration.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Should be a simple matter of overriding `class CreateTableStatementSegment` in the dialect_clickhouse.py file and adding a test case as detailed in [Contributing Dialect Changes wiki](https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes) if anyone fancies a stab at this.

**Comment by tomasfarias** [comment]
Hey both :wave: I'm starting a new job working with ClickHouse, and I'd like to bring `sqlfluff` with me. As I'm fairly new to ClickHouse, I thought it would be a good learning experience to work on this while also contributing to `sqlfluff` support for ClickHouse.

Any comments on the linked PR are quite welcome!
